### PR TITLE
[rom_ctrl,sival] Add sival description for rom_ctrl block

### PR DESCRIPTION
**Pull request by rswarbrick**

### CODE REVIEW

File: hw/ip/rom_ctrl/data/rom_ctrl.hjson [hunk 1]
```diff
@@ -73,6 +73,27 @@
...
+    },
+    { name: "ROM_CTRL.DIGESTS",
+      desc: '''
+        A signature of the ROM contents is computed at startup.
+        This is available in the DIGEST_<i> CSRs.
+      '''
+    },
+    { name: "ROM_CTRL.EXP_DIGESTS",
+      desc: '''
+        The chip has an expected ROM signature.
```
Comment (marnovandermaas) [inline hw/ip/rom_ctrl/data/rom_ctrl.hjson:91]: Does the ROM control do anything with this expected signature? Or is it up to firmware to check that the digest and the expected digest match?

File: hw/top_earlgrey/data/ip/chip_rom_ctrl_testplan.hjson [hunk 2]
```diff
@@ -21,8 +23,11 @@
               the ROM. Verify that we can successfully power up the chip in this case.
             - In PROD LC state, verify that the pwrmgr does not fully power up if the computed
               digest does not match the top 8 words of the ROM.
+
```
Comment (marnovandermaas) [inline hw/top_earlgrey/data/ip/chip_rom_ctrl_testplan.hjson:26]: Maybe add some notes for silicon targets that the prod modes can be tested in non-secure environments and non-prod stages will need to be done in a secure site.

**Review by marnovandermaas — APPROVED** [review]
Some comments from my side but in general looks good to me.

File: hw/ip/rom_ctrl/data/rom_ctrl.hjson (hunk 1 above)
Comment (rswarbrick) [reply hw/ip/rom_ctrl/data/rom_ctrl.hjson:91]: Yep! It blocks boot if the two don't match. The idea is that an evil attacker would need to somehow subvert the machinery that stops the chip from starting at all, and would also find that the ROM signature gets incorporated into the creator root identity (not by firmware), so the chip wouldn't have a recognised identity anyway.

File: hw/top_earlgrey/data/ip/chip_rom_ctrl_testplan.hjson (hunk 2 above)
Comment (rswarbrick) [reply hw/top_earlgrey/data/ip/chip_rom_ctrl_testplan.hjson:26]: Hmm, it's worse than that: something has gone horribly wrong if the expected and measured digests don't match (and we can probably give up on bring-up at that point!)

I guess it's still worth checking that the digests are sensibly readable by software. I'll add some notes about that and re-push. Thanks!

**Pull request merged**